### PR TITLE
feat: add-inv-functions

### PR DESCRIPTION
**Pull request by andybeltz**
### Summary

- Added `New-AriaNetworksvCenterDataSource` cmdlet to add a vCenter Server in VMware Aria Operations for Networks.
- Added `New-AriaNetworksNsxtDataSource` cmdlet to add a NSX Manager in VMware Aria Operations for Networks.

### Type

- [ ] Bugfix
- [X] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### Test and Documentation

- [X] Tests have been completed.
- [x] Documentation has been added or updated.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$id,
+		[Parameter (Mandatory = $true)] [ValidateSet('vcenter', 'nsxt')] [ValidateNotNullOrEmpty()] [String]$dataSourceType
+    )
+
+    Try {
+        if ($ariaNetworksAppliance) {
+            if ($dataSourceType -eq 'vcenter') {
+                $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/vcenters/$id"
+                (Invoke-RestMethod -Method 'DEL' -Uri $uri -Headers $ariaNetworksHeader).results
+            } elseif($dataSourceType -eq 'nsxt') {
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50320]:
```suggestion
            } elseif ($dataSourceType -eq 'nsxt') {
```

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+    } Catch {
+        Write-Error $_.Exception.Message
+    }
+}
+Export-ModuleMember -Function Remove-AriaNetworksDataSource
+
+Function New-AriaNetworksvCenterDataSource {
+    <#
+        .SYNOPSIS
+        Add a new vCenter Data Source to VMware Aria Operations for Networks.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50336]:
```suggestion
        Add a new vCenter Server data source to VMware Aria Operations for Networks.
```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+
+        .DESCRIPTION
+        The New-AriaNetworksvCenterDataSource cmdlet allows a user to add in a vCenter Server as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50339]:
```suggestion
        The New-AriaNetworksvCenterDataSource cmdlet allows a user to add a vCenter Server as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+
+        .EXAMPLE
+        New-AriaNetworksvCenterDataSource -fqdn sfo-m01-vc01.sfo.rainpole.io -username svc-inv-vsphere -password VMw@re1! -nickname "sfo-m01-vc01 - Management Domain vCenter Server" -CollectorId 15832:901:1711011916294613031 -enabled true
+        This example adds a vCenter server as a new data source in VMware Aria Operations for Networks.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50343]:
```suggestion
        This example adds a vCenter Server as a new data source in VMware Aria Operations for Networks.
```

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+        The username to use for authentication.
+
+        .PARAMETER password
+        The password to use for authentication.
+
+        .PARAMETER nickname
+        The nickname to use for this data source in VMware Aria Operations for Networks.
+
+        .PARAMETER collectorId
+        The id of the VMware Aria Operations for Networks collector node where the vCenter data source will be connected.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50358]:
```suggestion
        The id of the VMware Aria Operations for Networks collector node where the vCenter Server data source will be connected.
```

File: PowerValidatedSolutions.psm1 [hunk 6]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50361]:
```suggestion
        The parameter to enable the data source.
```

File: PowerValidatedSolutions.psm1 [hunk 7]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+    #>
+
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50370]:
```suggestion
      [Parameter (Mandatory = $true)] [ValidateSet('true', 'false')] [String]$enabled
```

File: PowerValidatedSolutions.psm1 [hunk 8]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+    )
+
+    Try {
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50373]: Any reason your not consistently using the if ($ariaNetworksAppliance) { } else { Write-Error ""} consistently

File: PowerValidatedSolutions.psm1 [hunk 9]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+  } Catch {
+        Write-Error $_.Exception.Message
+  }
+}
+Export-ModuleMember -Function New-AriaNetworksvCenterDataSource
+
+Function New-AriaNetworksNsxtDataSource {
+    <#
+        .SYNOPSIS
+        Add a new NSX-T Data Source to VMware Aria Operations for Networks.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50398]:
```suggestion
        Add a new NSX-T data source to VMware Aria Operations for Networks.
```

File: PowerValidatedSolutions.psm1 [hunk 10]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+
+        .DESCRIPTION
+        The New-AriaNetworksNsxtDataSource cmdlet allows a user to add in a NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50401]:
```suggestion
        The New-AriaNetworksNsxtDataSource cmdlet allows a user to add an NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```

File: PowerValidatedSolutions.psm1 [hunk 11]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50447]:
```suggestion
      [Parameter (Mandatory = $true)] [ValidateSet('true', 'false')] [String]$enabled
```

File: PowerValidatedSolutions.psm1 [hunk 12]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+        $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/nsxt-managers"
+        if ($PSBoundParameters.ContainsKey('certificate') -and $PSBoundParameters.ContainsKey('privatekey')) {
+              if (Test-Path -Path $certificate) {
+                $cert = (Get-Content -Path $certificate) -join "`n"
+              }
+              else {Write-Error "Unable to find certificate file ($certificate)"}
+              if (Test-Path -Path $privatekey) {
+                $key = (Get-Content -Path $privatekey) -join "`n"
+              }
+              else {Write-Error "Unable to find key file ($privatekey)"}
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50460]:
```suggestion
              } else {
                  Write-Error "Unable to find key file ($privatekey)"}
```

File: PowerValidatedSolutions.psm1 [hunk 13]
```diff
@@ -50287,6 +50287,208 @@ Function Get-AriaNetworksDataSource {
...
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+    )
+
+    Try {
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:50450]: Any reason your not consistently using the if ($ariaNetworksAppliance) { } else { Write-Error ""} consistently

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md [hunk 14]
```diff
@@ -0,0 +1,177 @@
+# New-AriaNetworksNsxtDataSource
+
+## Synopsis
+
+Add a new NSX-T Data Source to VMware Aria Operations for Networks.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:5]:
```suggestion
Add a new NSX-T data source to VMware Aria Operations for Networks.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+### certificate
+
+```powershell
+New-AriaNetworksNsxtDataSource -fqdn <String> [-certificate <String>] [-privatekey <String>] -nickname <String>
+ -collectorId <String> -enabled <String> [<CommonParameters>]
+```
+
+## Description
+
+The `New-AriaNetworksNsxtDataSource` cmdlet allows a user to add in a NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:25]:
```suggestion
The `New-AriaNetworksNsxtDataSource` cmdlet allows a user to add an NSX-T Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+
+## Examples
+
+### Example 1
+
+```powershell
+New-AriaNetworksNsxtDataSource -fqdn sfo-m01-nsx01.sfo.rainpole.io -certificate F:\certs\sfo-m01-nsx01.cer -privatekey F:\certs\sfo-m01-nsx01.key -nickname "sfo-m01-nsx01 - Management Domain NSX Manager" -CollectorId 15832:901:1711011916294613031 -enabled true
+```
+
+This example adds a NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSXT-T principal identity user.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:35]:
```suggestion
This example adds an NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSXT-T principal identity user.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md [hunk 17]
```diff
@@ -0,0 +1,128 @@
+# New-AriaNetworksvCenterDataSource
+
+## Synopsis
+
+Add a new vCenter Data Source to VMware Aria Operations for Networks.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md:5]:
```suggestion
Add a new vCenter Server data source to VMware Aria Operations for Networks.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+## Syntax
+
+```powershell
+New-AriaNetworksvCenterDataSource [-fqdn] <String> [-username] <String> [-password] <String>
+ [-nickname] <String> [-collectorId] <String> [-enabled] <String> [<CommonParameters>]
+```
+
+## Description
+
+The `New-AriaNetworksvCenterDataSource` cmdlet allows a user to add in a vCenter Server as a new data source in VMware Aria Operations for Networks in order to collect network flow data.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md:16]:
```suggestion
The `New-AriaNetworksvCenterDataSource` cmdlet allows a user to add a vCenter Server as a new data source in VMware Aria Operations for Networks in order to collect network flow data.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/Remove-AriaNetworksDataSource.md [hunk 19]
```diff
@@ -0,0 +1,80 @@
+# Remove-AriaNetworksDataSource
+
+## Synopsis
+
+Remove a data source from a VMware Aria Operations for Networks deployment.
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-operations-networks/Remove-AriaNetworksDataSource.md:5]:
```suggestion
Remove a data source from VMware Aria Operations for Networks .

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md (hunk 16 above)
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:35]:
```suggestion
This example adds an NSX-T Manager server as a new data source in VMware Aria Operations for Networks by using a NSX principal identity user.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md (hunk 15 above)
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:25]:
```suggestion
The `New-AriaNetworksNsxtDataSource` cmdlet allows a user to add an NSX Manager as a new data source in VMware Aria Operations for Networks in order to collect network flow data.

```

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md (hunk 16 above)
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:35]: Could we use "NSX" instead of NSX-T now?

Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:35]: Function names (`...Nsxt...`) could remain.

File: PowerValidatedSolutions.psm1 [hunk 20]
```diff
@@ -3737,7 +3737,7 @@ Function Invoke-PdrSolutionInterop {
                                             messageHandler -statusMessage $StatusMsg -warningMessage $WarnMsg -errorMessage $ErrorMsg; if ($ErrorMsg) { $failureDetected = $true }
                                             $StatusMsg = Add-vROPSAdapterVr -server $jsonInput.protected.sddcManagerFqdn -user $jsonInput.protected.sddcManagerUser -pass $jsonInput.protected.sddcManagerPass -vrFqdn $jsonInput.recovery.vrmsFqdn -vrUser ($jsonInput.recovery.ssoServiceAccountVrmsUser + "@vsphere.local") -vrPass $jsonInput.recovery.ssoServiceAccountVrmsPass -collectorGroupName $jsonInput.recovery.collectorGroup -WarningAction SilentlyContinue -ErrorAction SilentlyContinue -WarningVariable WarnMsg -ErrorVariable ErrorMsg
                                             messageHandler -statusMessage $StatusMsg -warningMessage $WarnMsg -errorMessage $ErrorMsg; if ($ErrorMsg) { $failureDetected = $true }
-                                            
+
                                         }
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:3741]:
```suggestion
                                        }
```

File: PowerValidatedSolutions.psm1 [hunk 21]
```diff
@@ -51303,7 +51303,6 @@ Function Get-AriaNetworksDataSource {
     }
 }
 Export-ModuleMember -Function Get-AriaNetworksDataSource
-
 Function Remove-AriaNetworksDataSource {
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51306]:
```suggestion

Function Remove-AriaNetworksDataSource {
```

File: PowerValidatedSolutions.psm1 [hunk 22]
```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
...
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
     )
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51387]:
```suggestion
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
    )
```

File: PowerValidatedSolutions.psm1 [hunk 23]
```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
...
+        .PARAMETER nickname
+        The nickname to use for this data source in VMware Aria Operations for Networks.
+
+        .PARAMETER collectorId
+        The id of the VMware Aria Operations for Networks collector node where the NSX-T Manager data source will be connected.
+
+        .PARAMETER enabled
+        The parameter to enable the datasource.
+
+    #>
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51451]:
```suggestion
        The parameter to enable the datasource.
    #>
```

File: PowerValidatedSolutions.psm1 [hunk 24]
```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
...
+            ipfix_request      = @{
+              "enable_all"= $true
+            }
+          } | ConvertTo-Json
+        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
   } Catch {
         Write-Error $_.Exception.Message
   }
 }
 Export-ModuleMember -Function New-AriaNetworksvCenterDataSource
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51409]:
```suggestion
        $body = @{
            fqdn = $fqdn
            proxy_id = $collectorId
            nickname = $nickname
            enabled = $enabled
            credentials = @{
                username = $username
                password = $password
            }
            ipfix_request = @{
                "enable_all"= $true
            }
        } | ConvertTo-Json
        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
    } Catch {
        Write-Error $_.Exception.Message
    }
}
```

File: PowerValidatedSolutions.psm1 [hunk 25]
```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
...
+    Param (
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
+      [Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+    )
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51464]:
```suggestion
    Param (
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
		[Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$certificate,
		[Parameter (Mandatory = $false, ParameterSetName = 'certificate')] [ValidateNotNullOrEmpty()] [String]$privatekey,
		[Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$username,
		[Parameter (Mandatory = $false, ParameterSetName = 'credentials')] [ValidateNotNullOrEmpty()] [String]$password,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
    )
```

File: PowerValidatedSolutions.psm1 [hunk 26]
```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
...
+                  username = $username
+                  password = $password
+                }
+              } | ConvertTo-Json
+          }
+            Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
+  } Catch {
+        Write-Error $_.Exception.Message
+  }
+}
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:51506]:
```suggestion
    Try {
        $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/nsxt-managers"
        if ($PSBoundParameters.ContainsKey('certificate') -and $PSBoundParameters.ContainsKey('privatekey')) {
            if (Test-Path -Path $certificate) {
                $cert = (Get-Content -Path $certificate) -join "`n"
            } else {
                Write-Error "Unable to find certificate file ($certificate)"
            }
            if (Test-Path -Path $privatekey) {
                $key = (Get-Content -Path $privatekey) -join "`n"
            } else {
                Write-Error "Unable to find key file ($privatekey)"
            }
            $body = @{
                fqdn = $fqdn
                proxy_id = $collectorId
                nickname = $nickname
                enabled = $enabled
                ipfix_enabled = $true
                latency_enabled = $true
                cred_type = "CERTIFICATE"
                client_certificate = $cert
                client_private_key = $key
            } | ConvertTo-Json
        } elseif ($PSBoundParameters.ContainsKey('username') -and $PSBoundParameters.ContainsKey('password')) {
            $body = @{
                fqdn = $fqdn
                proxy_id = $collectorId
                nickname = $nickname
                enabled = $enabled
                ipfix_enabled = $true
                latency_enabled = $true
                credentials = @{
                    username = $username
                    password = $password
                }
            } | ConvertTo-Json
        }
        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
    } Catch {
        Write-Error $_.Exception.Message
    }
}
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Some formatting updates.

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md (hunk 15 above)
Comment (andybeltz) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md (hunk 16 above)
Comment (andybeltz) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksNsxtDataSource.md:39]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 24 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 26 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 25 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 23 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 22 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 21 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 20 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md (hunk 17 above)
Comment (andybeltz) [reply docs/documentation/functions/aria-suite/aria-operations-networks/New-AriaNetworksvCenterDataSource.md]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80

File: PowerValidatedSolutions.psm1 (hunk 13 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/5b6f2b46e0444235d6f3d74157fc959015d63688

File: PowerValidatedSolutions.psm1 (hunk 8 above)
Comment (andybeltz) [reply PowerValidatedSolutions.psm1]: Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/5b6f2b46e0444235d6f3d74157fc959015d63688

**Pull request merged**